cradle-core/src/main/java/com/exactpro/cradle/serialization/MessageDeserializer.java

```diff
@@ -0,0 +1,136 @@
+	}
+	
+	public List<StoredMessage> deserializeBatch(byte[] buffer) throws SerializationException {
+		return this.deserializeBatch(ByteBuffer.wrap(buffer));
+	}
+	
+	private void checkMessageBatchMagics(ByteBuffer buffer) throws SerializationException {
+		int magicNumber = buffer.getInt();
+		if (magicNumber != MESSAGE_BATCH_MAGIC) {
+			throw SerializationUtils.incorrectMagicNumber("MESSAGE_BATCH", magicNumber, MESSAGE_BATCH_MAGIC);
+		}
+
+		byte protocolVer = buffer.get();
+		if (protocolVer != MESSAGE_PROTOCOL_VER) {
+			throw new SerializationException(String.format(NOT_SUPPORTED_PROTOCOL_FORMAT, "message batches",
+					protocolVer, MESSAGE_PROTOCOL_VER));
+		}
+	}
+	
+	public List<StoredMessage> deserializeBatch(ByteBuffer buffer) throws SerializationException {
+		checkMessageBatchMagics(buffer);
+
+		int messagesCount = buffer.getInt();
+		List<StoredMessage> messages = new ArrayList<>(messagesCount);
+		for (int i = 0; i < messagesCount; ++i) {
+			int msgLen = buffer.getInt();
+			ByteBuffer msgBuf = ByteBuffer.wrap(buffer.array(), buffer.position(), msgLen);
+			messages.add(this.deserialize(msgBuf));
+			buffer.position(buffer.position() + msgLen);
+		}
+		
+		return messages;
+	}
+
+	public StoredMessage deserializeOneMessage(ByteBuffer buffer, StoredMessageId id) throws SerializationException {
+		checkMessageBatchMagics(buffer);
+
+		int messagesCount = buffer.getInt();
+		for (int i = 0; i < messagesCount; ++i) {
+			int msgLen = buffer.getInt();
+			ByteBuffer msgBuf = ByteBuffer.wrap(buffer.array(), buffer.position(), msgLen);
+			StoredMessage msg = this.deserialize(msgBuf);
+			if (msg.getId().equals(id)) {
+				return msg;
+			}
+			buffer.position(buffer.position() + msgLen);
+		}
+
+		return null;
+	}
+	
+	public StoredMessage deserialize(byte[] message) throws SerializationException {
+		return deserialize(ByteBuffer.wrap(message));
+	}
+
+	public StoredMessage deserialize(ByteBuffer buffer) throws SerializationException {
+		short magicNumber = buffer.getShort();
+		if (magicNumber != MESSAGE_MAGIC) {
+			throw SerializationUtils.incorrectMagicNumber(StoredMessage.class.getSimpleName(), magicNumber, MESSAGE_MAGIC);
+		}
+		StoredMessageBuilder builder = new StoredMessageBuilder();
+		builder.setMessageId(readMessageId(buffer));
+		builder.setTimestamp(readInstant(buffer));
+		readMessageMetaData(buffer, builder);
+		builder.setContent(readBody(buffer));
+		return builder.build();
+	}
+	
+	private Direction getDirection(int ordinal) throws SerializationException {
+		Direction[] values = Direction.values();
+		if (values.length > ordinal && ordinal >= 0)
+			return values[ordinal];
+		throw new SerializationException(String.format("Invalid ordinal for enum (Direction): %d. Values: [0-%d]",
+				ordinal, values.length - 1));
+	}
+
+	private StoredMessageId readMessageId(ByteBuffer buffer) throws SerializationException {
+		String stream = readShortString(buffer);
+		Direction direction = getDirection(buffer.get());
```

Comment:
I'm sure you can omit saving of this for all messages in batch. These two values should be saved only once per batch.